packages/cli/src/util/__tests__/paths.spec.ts

```diff
@@ -0,0 +1,189 @@
+import { assertLeft, assertRight } from '@stoplight/prism-http/src/__tests__/utils';
+import { HttpParamStyles } from '@stoplight/types/dist';
+import { createExamplePath } from '../paths';
+
+describe('createExamplePath()', () => {
+  describe('path parameters', () => {
+    it('generates simple style', () => {
+      assertRight(
+        createExamplePath({
+          id: '123',
+          path: '/path/{p}',
+          method: 'get',
+          request: { path: [{ name: 'p', style: HttpParamStyles.Simple, examples: [{ key: 'foo', value: 'test' }] }] },
+          responses: [{ code: '200' }],
+        }),
+        r => expect(r).toEqual('/path/test'),
+      );
+    });
+
+    it('generates label style', () => {
+      assertRight(
+        createExamplePath({
+          id: '123',
+          path: '/path/{p}',
+          method: 'get',
+          request: { path: [{ name: 'p', style: HttpParamStyles.Label, examples: [{ key: 'foo', value: 'test' }] }] },
+          responses: [{ code: '200' }],
+        }),
+        r => expect(r).toEqual('/path/.test'),
+      );
+    });
+
+    it('generates matrix style', () => {
+      assertRight(
+        createExamplePath({
+          id: '123',
+          path: '/path/{p}',
+          method: 'get',
+          request: { path: [{ name: 'p', style: HttpParamStyles.Matrix, examples: [{ key: 'foo', value: 'test' }] }] },
+          responses: [{ code: '200' }],
+        }),
+        r => expect(r).toEqual('/path/;p=test'),
+      );
+    });
+
+    it('fails when not supported style supplied', () => {
+      assertLeft(
+        createExamplePath({
+          id: '123',
+          path: '/path/{p}',
+          method: 'get',
+          request: {
+            path: [
+              { name: 'p', style: HttpParamStyles.PipeDelimited as any, examples: [{ key: 'foo', value: 'test' }] },
```

Comment:
The good thing of TypeScript is that it will avoid us to make some troubles and so we can avoid both the runtime checks and testing for such conditions. I'd remove this test completely and the runtime checks in the function as well.